core/src/core/transaction.rs

```diff
@@ -839,10 +858,11 @@ impl ops::Add for SumCommit {
 			Ok(s) => s,
 			Err(_) => Commitment::from_vec(vec![1; 33]),
 		};
-		SumCommit::new(
-			self.features | other.features,
-			&sum,
-		)
+		SumCommit {
+			features: self.features | other.features,
+			commit: sum,
+			switch_commit_hash: SwitchCommitHash::zero(),
```

Comment:
I think that'd be better as the hash of the 2 commit hashes, so that it can roll up the tree as well.